src/Microsoft.Health.Client/OAuth2ClientCredentialProvider.cs

```diff
@@ -46,9 +46,13 @@ protected override async Task<string> BearerTokenFunction(CancellationToken canc
             };
 
             using var formContent = new FormUrlEncodedContent(formData);
-            using HttpResponseMessage tokenResponse = await _httpClient.PostAsync(_oAuth2ClientCredentialConfiguration.TokenUri, formContent, cancellationToken);
+            using HttpResponseMessage tokenResponse = await _httpClient.PostAsync(_oAuth2ClientCredentialConfiguration.TokenUri, formContent, cancellationToken).ConfigureAwait(false);
 
-            var openIdConnectMessage = new OpenIdConnectMessage(await tokenResponse.Content.ReadAsStringAsync());
+#if NET5_0_OR_GREATER
```

Comment:
Ugh.

Unfortunately only .NET 5 has the overload where we can propagate the token